pkg/rbac/middleware.go

```diff
@@ -5,19 +5,42 @@ import (
 	"encoding/json"
 	"net/http"
 	"strings"
+	"sync"
+	"time"
 
 	"github.com/stolostron/search-v2-api/pkg/config"
 	authv1 "k8s.io/api/authentication/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/klog/v2"
 )
 
+//SAVING THIS IN ONLY BRANCH ANX/AUTHENTICATE
+
+//first we want to cache the authenticated tokenreview response:
+type UserTokenReviews struct {
```

Comment:
Can we rename this to `UserData` ?
`UserTokenReviews` is a little confusing because we will be adding more data later.